### DISCUSSION TITLE
0.2.2

### DISCUSSION THREAD

**Discussion by kubilus1** [kubilus1/autoortho#30] [Announcements]
This release has a few important changes:

* The cache directory is now configurable and defaults to your user home directory.  Existing cache folders can be preserved by moving existing cache files into this new folder using standard filesystem management tools.
* Lots more checks and bug fixes around scenery setup
* A stats page and flight map is available via your web browser.  Point to port 5000 from wherever you run XPlane.  (So http://localhost:5000 if running XPlane locally)
* New configuration options are available
* New sceneries are now supported.  South America has been added but Asia and Africa will be available soon.

## What's Changed
* Add more config options, stats pages, and more checking around downloads by @kubilus1 in https://github.com/kubilus1/autoortho/pull/26
* Update docs by @kubilus1 in https://github.com/kubilus1/autoortho/pull/27
* Pull live config by @kubilus1 in https://github.com/kubilus1/autoortho/pull/28
* Fix tests, add more download checking by @kubilus1 in https://github.com/kubilus1/autoortho/pull/29

**Full Changelog**: https://github.com/kubilus1/autoortho/compare/0.2.1...0.2.2

<hr /><em>This discussion was created from the release <a href='https://github.com/kubilus1/autoortho/releases/tag/0.2.2-2'>0.2.2</a>.</em>

**Comment by ivotaHR** [comment]
hi !!

root: H:/X-Plane 12/Custom Scenery\z_autoortho\_textures
mountpoint: H:/X-Plane 12/Custom Scenery\z_autoortho\textures
Creating stats object
Starting stats thread
Running in Windows WinFSP mode.
Windows detected
chunk_getter: <getortho.ChunkGetter object at 0x000001DE003D3D90>
Traceback (most recent call last):
  File "C:\Users\Ivica\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Ivica\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "F:\autoortho-0.2.2\autoortho\__main__.py", line 18, in <module>
    autoortho.main()
  File "F:\autoortho-0.2.2\autoortho\autoortho.py", line 80, in main
    import autoortho_winfsp
  File "F:\autoortho-0.2.2\autoortho\autoortho_winfsp.py", line 34, in <module>
    from autoortho import CFG
ImportError: cannot import name 'CFG' from 'autoortho' (F:\autoortho-0.2.2\autoortho\autoortho.py)
STATS: ID: {}
STATS: ID: {}

**Reply by lugs5678** [reply]
same thing here

**Reply by kubilus1** [reply]
Ah, I see the issue.  Fixed!

**Reply by kubilus1** [reply]
I may need to do one more tweak to this release.  I'll update shortly.

**Comment by ivotaHR** [comment]
fix !! its working.

Info:

Python python-3.10.9-amd64 ->> working OK
python-3.11.1-amd64 -->> NOT working (winfsp problem install)

**Reply by kubilus1** [reply]
FYI: The python 3.11 with winfsp install is resolvable, but you will need to follow the suggestions in the error that's shown.

**Reply by ivotaHR** [reply]
I am staying on 3.10.x

**Comment by ivotaHR** [comment]
BTW, Thank you for a new Update.

**Comment by betampona** [comment]
Hi, I flew from Mendoza SAME to Santo Domigo in Chile SCSN with a Cirrus SF50 pushed to maximum. Nothing to complain about, keep it up.
I still have to test with the zibo.
![CirrusSF50 - 2022-12-28 13 26 58](https://user-images.githubusercontent.com/97102929/209814412-bbfe5e47-8405-4bcf-8be2-55aa0a6dbbbb.png)

**Comment by kubilus1** [comment]
Great.  Making this release the official latest release now.

**Comment by lugs5678** [comment]
Is it possible to optimise the CPU usage in this, im on a Ryzen 5 5600X and it it will sometimes shoot up in usage at cruise and cause major fram drops.

**Reply by kubilus1** [reply]
CPU usage should be pretty well optimized.  This uses a compressor built with ISPC and with AVX extensions.  Probably not a lot of more room on that front.

**Reply by lugs5678** [reply]
any idea why it causes huge CPU spikes sometimes? would using a map override option help, like setting it to ARC or Bing?

**Reply by kubilus1** [reply]
When you are getting close to a DSF boundary, XPlane is attempting to pull in many many tiles at the same time, which leads to spikes.  Bing is the default for the maps I've generated.  Imagery source shouldn't make any different for CPU processing.

You could try setting the compressor type in the config file to STB instead of ISPC and see if that makes a difference, but likely that will just be worse.

**Comment by betampona** [comment]
I have just made a flight with the Zibo, FL 350 speed 460 Kts during 1 hour on the Brazilian no blue tiles, for me it can be exploitable in IFR with planes liners.
I will continue to do tests this week.
To have this quality I am now in field of view of 60° which is the native value of X-plane11.
Can you include a desktop shortcut with an icon.
Thanks for the work done, you are good at programming.

Translated with www.DeepL.com/Translator (free version)

**Comment by ivotaHR** [comment]
>  Can you include a desktop shortcut with an icon.

Make icon on desktop by yourself. Mouse click on run.bat and **sent to** shortcut to desktop.  then on created desktop icon, properties , Change Icon, choose the icon what you wish.

**Reply by betampona** [reply]
I was talking about an icon in the color of autoortho as there is one in xplane

**Reply by betampona** [reply]
Here is a simple example of an icon:
![AO icone (2)](https://user-images.githubusercontent.com/97102929/209909279-3e380441-469b-4dad-96d2-e4f589492126.png)

**Comment by betampona** [comment]
I can find out how this option is used and show us a picture of how it looks:
A stats page and flight map is available via your web browser. Point to port 5000 from wherever you run XPlane. (So http://localhost:5000/ if running XPlane locally)

**Comment by betampona** [comment]
Hello, I am having trouble viewing the statics page and the flight map.
I have set up x-plane 11 see photo, I launch via google chrome http://localhost:5000 and I have the message ERR_CONNECTION_REFUSED
What is the method to view the pages?

![auto 1](https://user-images.githubusercontent.com/97102929/210132323-76ba5fbd-2841-4dd7-bf6e-4a9a1d4c4c2a.JPG)

**Reply by kubilus1** [reply]
You should leave all the port options in XPlane as default. 

As far as seeing the stats/map page it will be wherever you run XPlane from at port 5000.  So it looks like in your case you could try 'http://192.168.1.31:5000' from your web browser of choice.

**Comment by betampona** [comment]
Hello, I'm still blocked to get the statistics and the map. In the task manager I have
lkcitdl.exe with port 5000 protocol status not allowed

**Comment by ivotaHR** [comment]
I am flying World IFR tour. Can you check PHNL (no ortho).

**Comment by ivotaHR** [comment]
![1234](https://user-images.githubusercontent.com/96768739/216826179-bfb483c9-6e90-4464-bd0e-054cd34391b9.JPG)

Missing AU.

**Comment by kubilus1** [comment]
Good catch.  Corrected.  Thanks!

**Comment by ivotaHR** [comment]
Failed getting: Chunk(38214,27184,BI,16,16) () {}, re-submit.
Failed to get chunk Chunk(4773,3401,BI,13,13) on server a. Err: <urlopen error timed out>
Failed getting: Chunk(4773,3401,BI,13,13) () {}, re-submit.
CLOSE: Tile(38176, 27200, BI, 16, 12, F:\.autoortho-data\cache) REFS: 0
CLOSE: No refs.  Removing reference to tile: Tile(38176, 27200, BI, 16, 12, F:\.autoortho-data\cache)!
Open: Tile(38192, 27184, BI, 16, 12, F:\.autoortho-data\cache)
OPEN: Tile(38192, 27184, BI, 16, 12, F:\.autoortho-data\cache) REFS: 1
CLOSE: Tile(38192, 27184, BI, 16, 12, F:\.autoortho-data\cache) REFS: 0
CLOSE: No refs.  Removing reference to tile: Tile(38192, 27184, BI, 16, 12, F:\.autoortho-data\cache)!
Open: Tile(38192, 27184, BI, 16, 12, F:\.autoortho-data\cache)
OPEN: Tile(38192, 27184, BI, 16, 12, F:\.autoortho-data\cache) REFS: 1
STATS: ID: {'count': 1236, 'chunk_miss': 1236, 'bytes_dl': 17404892, 'mm_counts': {0: 1, 1: -1, 2: -1, 3: -1, 4: 27, 5: 31}, 'mm_averages': {0: 0.52, 1: -1, 2: -1, 3: -1, 4: 0.01, 5: 0.01}}
CLOSE: Tile(38048, 27040, BI, 16, 12, F:\.autoortho-data\cache) REFS: 0
CLOSE: No refs.  Removing reference to tile: Tile(38048, 27040, BI, 16, 12, F:\.autoortho-data\cache)!
Open: Tile(38064, 27040, BI, 16, 12, F:\.autoortho-data\cache)
OPEN: Tile(38064, 27040, BI, 16, 12, F:\.autoortho-data\cache) REFS: 1
CLOSE: Tile(38064, 27040, BI, 16, 12, F:\.autoortho-data\cache) REFS: 0
CLOSE: No refs.  Removing reference to tile: Tile(38064, 27040, BI, 16, 12, F:\.autoortho-data\cache)!
Open: Tile(38064, 27040, BI, 16, 12, F:\.autoortho-data\cache)
OPEN: Tile(38064, 27040, BI, 16, 12, F:\.autoortho-data\cache) REFS: 1
Failed to get chunk Chunk(38079,27088,BI,16,16) on server b. Err: <urlopen error timed out>
Failed to get chunk Chunk(38072,27088,BI,16,16) on server c. Err: <urlopen error timed out>
Failed to get chunk Chunk(38068,27088,BI,16,16) on server d. Err: <urlopen error timed out>
Failed to get chunk Chunk(38075,27088,BI,16,16) on server c. Err: <urlopen error timed out>
Failed getting: Chunk(38079,27088,BI,16,16) () {}, re-submit.
Failed getting: Chunk(38072,27088,BI,16,16) () {}, re-submit.
Failed getting: Chunk(38068,27088,BI,16,16) () {}, re-submit.
Failed getting: Chunk(38075,27088,BI,16,16) () {}, re-submit.
